nemo/collections/asr/data/data_simulation.py

```diff
@@ -1460,3 +1466,901 @@ def _generate_session(self, idx: int, basepath: str, filename: str, enforce_coun
+            ax.text(label_pos[0], label_pos[1], label_pos[2], label, tuple(self.internal_cs[dim]), c='blue')
+        try:
+            # Unfortunately, equal aspect ratio has been added very recently to Axes3D
+            ax.set_aspect('equal')
+        except NotImplementedError:
+            logging.warning('Equal aspect ratio not supported by Axes3D')
+        # Set view
+        ax.view_init(elev=elev, azim=azim)
+        # Set reasonable limits for all axes, even for the case of an unequal aspect ratio
+        ax.set_xlim([self.center[0] - self.radius, self.center[0] + self.radius])
+        ax.set_ylim([self.center[1] - self.radius, self.center[1] + self.radius])
+        ax.set_zlim([self.center[2] - self.radius, self.center[2] + self.radius])
+
+        ax.set_xlabel('x/m')
+        ax.set_ylabel('y/m')
+        ax.set_zlabel('z/m')
+        ax.set_title('Microphone positions')
+        ax.legend()
+        plt.show()
+
+
+def convert_placement_to_range(
+    placement: Dict, room_dim: Iterable[float], object_radius: float = 0
+) -> List[List[float]]:
+    """Given a placement dictionary, return ranges for each dimension.
+
+    Args:
+        placement: dictionary containing x, y, height, and min_to_wall
+        room_dim: dimensions of the room, shape (3,)
+        object_radius: radius of the object to be placed
+
+    Returns
+        List with a range of values for each dimensions.
+    """
+    if not np.all(np.array(room_dim) > 0):
+        raise ValueError(f'Room dimensions must be positive: {room_dim}')
+
+    placement_range = [None] * 3
+    min_to_wall = placement.get('min_to_wall', 0)
+
+    if min_to_wall < 0:
+        raise ValueError(f'Min distance to wall must be positive: {min_to_wall}')
+
+    for idx, key in enumerate(['x', 'y', 'height']):
+        # Room dimension
+        dim = room_dim[idx]
+        # Construct the range
+        val = placement.get(key)
+        if val is None:
+            # No constrained specified on the coordinate of the mic center
+            min_val, max_val = 0, dim
+        elif np.isscalar(val):
+            min_val = max_val = val
+        else:
+            if len(val) != 2:
+                raise ValueError(f'Invalid value for placement for dim {idx}/{key}: {str(placement)}')
+            min_val, max_val = val
+
+        # Make sure the array is not too close to a wall
+        min_val = max(min_val, min_to_wall + object_radius)
+        max_val = min(max_val, dim - min_to_wall - object_radius)
+
+        if min_val > max_val or min(min_val, max_val) < 0:
+            raise ValueError(f'Invalid range dim {idx}/{key}: min={min_val}, max={max_val}')
+
+        placement_range[idx] = [min_val, max_val]
+
+    return placement_range
+
+
+class RIRCorpusGenerator(object):
+    """Creates a corpus of RIRs based on a defined configuration of rooms and microphone array.
+
+    RIRs are generated using `generate` method.
+    """
+
+    def __init__(self, cfg: DictConfig):
```

Comment:
Hmm.. I think it would be better design to be explicit in this init, defining all params. Use hydra.utils.instantiate() to inflate from a config if necessary. Right now its opaque as to what exactly the config should contain.